File: packages/core/lib/src/entities/track.dart
class Track {
  final String id;
  final String filePath;
  final String title;
  final String artist;
  final Duration duration;
  final String? artworkUrl;
  final String? previewUrl;

  const Track({
    required this.id,
    required this.filePath,
    required this.title,
    required this.artist,
    required this.duration,
    this.artworkUrl,
    this.previewUrl,
  });
}

File: models/track.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(slots=True)
class Track:
    file_path: str
    title: str
    artist: str
    duration: float

    @property
    def duration_seconds(self) -> int:
        return max(0, int(round(self.duration)))

    @property
    def display_duration(self) -> str:
        minutes, seconds = divmod(self.duration_seconds, 60)
        return f"{minutes}:{seconds:02d}"
